pkg/hostagent/hostagent.go

```diff
@@ -357,6 +358,146 @@ func (a *HostAgent) processGuestAgentEvents(ctx context.Context, localUnix strin
 	return io.EOF
 }
 
+func forwardTCP(ctx context.Context, sshConfig *ssh.SSHConfig, port int, local, remote string, cancel bool) error {
+	// on macOS, listening on 127.0.0.1:80 requires root while 0.0.0.0:80 does not require root.
+	// https://twitter.com/_AkihiroSuda_/status/1403403845842075648
+	// We use "pseudoloopback" forwarder that listens on 0.0.0.0:80 but rejects connections from non-loopback src IP.
+	if runtime.GOOS == "darwin" {
+		localIPStr, localPortStr, err := net.SplitHostPort(local)
+		if err != nil {
+			return err
+		}
+		localIP := net.ParseIP(localIPStr)
+		localPort, err := strconv.Atoi(localPortStr)
+		if err != nil {
+			return err
+		}
+		if net.ParseIP("127.0.0.1").Equal(localIP) && localPort < 1024 {
```

Comment:
I'm confused that there is no "else" branch for this condition, just a `return nil`. Doesn't that mean non-localhost, or non-privileged ports are no longer forwarded at all?

Reply:
So I think this should have a short block:

```
if !net.ParseIP("127.0.0.1").Equal(localIP) || localPort >= 1024 {
    return forwardSSH(ctx, sshConfig, port, local, remote, cancel)
}

and then the rest of the code could be outdented one level again.
```

Reply:
fixed